samples/bluetooth/hci_uart/overlay-all.conf

```diff
@@ -0,0 +1,69 @@
+CONFIG_BT_BUF_EVT_RX_COUNT=16
```

Comment:
Just out of curiosity: What does `all` refer her to here? All controller features?

Can we ensure that this is true, even when new features are added?

Reply:
We could at consider naming this file `overlay-all-bt_ll_sw_split.conf` as it looks like it contains controller specific configurations

Reply:
> Just out of curiosity: What does `all` refer her to here? All controller features?
> 
> Can we ensure that this is true, even when new features are added?

All conf files are manual process, updated by contributors. No, conf files can not be used to ensure new features are added. Atleast this file will be build tested in CI and will catch any config be removed.

I just followed https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/shell/shell/shell_min_cmds_all.conf, i assume is updated manually when adding new features.

Reply:
> We could at consider naming this file `overlay-all-bt_ll_sw_split.conf` as it looks like it contains controller specific configurations

done.

Reply:
> All conf files are manual process, updated by contributors. No, conf files can not be used to ensure new features are added. Atleast this file will be build tested in CI and will catch any config be removed.

IIRC when adding the new LLCP, there was a script to generate all possible mutations of several Kconfigs for testing building them. 

In theory it is possible to recreate something similar for this, where all Kconfigs that are `default n` to be enabled, so it should be possible. 

What the goal of this PR? Just to build a sample with them, or is it to actually run the sample with them as well?